x-pack/filebeat/input/httpjson/request.go

```diff
@@ -308,14 +327,16 @@ func newRequestFactory(ctx context.Context, config config, log *logp.Logger, met
 		var rf *requestFactory
 		// chain calls requestFactory object
 		if ch.Step != nil {
-			ts, _ := newBasicTransformsFromConfig(registeredTransforms, ch.Step.Request.Transforms, requestNamespace, log)
+			ts, _ := newBasicTransformsFromConfig(registeredTransforms, ch.Step.Request.Transforms, requestNamespace, stat, log)
 			ch.Step.Auth = tryAssignAuth(config.Auth, ch.Step.Auth)
-			client, err := newChainHTTPClient(ctx, ch.Step.Auth, ch.Step.Request, log, reg)
+			client, err := newChainHTTPClient(ctx, ch.Step.Auth, ch.Step.Request, stat, log, reg)
 			if err != nil {
-				return nil, fmt.Errorf("failed in creating chain http client with error: %w", err)
+				err = fmt.Errorf("failed in creating chain http client with error: %w", err)
+				stat.UpdateStatus(status.Degraded, err.Error())
```

Comment:
Does **Degraded** make sense here ? Any error in `newRequestFactory()` would bubble upto to run() and exit.

Reply:
Good point. Removed here since it will be handled by `run`.